docusaurus/docs/tutorials/build-an-app-plugin.md

```diff
@@ -0,0 +1,202 @@
+---
+id: build-an-app-plugin
+title: Build an app plugin
+description: Learn how to build an app plugin.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - app
+  - apps
+---
+
+import CreatePlugin from '@shared/create-plugin-frontend.md';
+import PluginAnatomy from '@shared/plugin-anatomy.md';
+
+# Build an app plugin
+
+App plugins are Grafana plugins that can bundle data source and panel plugins within one package. They also let you create _custom pages_ within Grafana. Custom pages enable the plugin author to include things like documentation, sign-up forms, or to control other services over HTTP.
+
+Data source and panel plugins show up like normal plugins. The app pages are available in Grafana's main menu.
+
+## Prerequisites
+
+- Grafana version 7.0 or later.
+- NodeJS 12.x or later.
+- yarn
+
+## Create a new plugin
+
+<CreatePlugin pluginType="app" />
+
+## Anatomy of a plugin
+
+<PluginAnatomy />
+
+## App plugins
+
+App plugins let you bundle resources such as dashboards, panels, and data sources into a single plugin.
+
+Any resource you want to include needs to be added to the `includes` property in the `plugin.json` file. To add a resource to your app plugin, you need to include it to the `plugin.json`.
+
+Plugins that are included in an app plugin are available like any other plugin.
+
+Dashboards and pages can be added to the app menu by setting `addToNav` to `true`.
+
+By setting `"defaultNav": true`, users can navigate to the dashboard by clicking the app icon in the side menu.
+
+### Add a custom page
+
+App plugins let you extend the Grafana user interface through the use of _custom pages_.
+
+Any requests sent to `/a/<plugin-id>`, e.g. `/a/myorgid-simple-app/`, are routed to the _root page_ of the app plugin. The root page is a React component that returns the content for a given route.
+
+While you're free to implement your own routing, in this tutorial you'll use a tab-based navigation page that you can use by calling `onNavChange`.
```

Comment:
When creating a new app plugin with `create-plugin`, we propose using `react-router-dom` rather than calling `onNavChange`. Since this is the first tutorial on app plugins, I wonder if it's better to describe the react-router-dom approach instead?